### ISSUE TITLE
gall: remote scries targeting tombstones hang forever

### ISSUE THREAD

**Issue by del-distributed** [urbit/urbit#6776]
TLDR: Remote scries targeting tombstones hang forever. The issue is addressed in #6743, but the problem persists for me after recompiling clay with the change. I propose #6778 as a partial fix for now.

**To Reproduce**
1. Recompile clay with #6743
2. Register a remote scry path
3. Tombstone it
4. Request it from a secondary ship

**Additional context**

Gall handles `%x` reads as follows:
```
?~  yok=(~(get by yokes.state) q.bem)  ~
    ?:  ?=(%nuke -.u.yok)  ~
    =/  ski  (~(get by sky.u.yok) path)
    ?~  ski  ~
    =/  res=(unit (each page @uvI))
      ?+    -.r.bem  ~
          %ud  (bind (get:on-path fan.u.ski p.r.bem) tail)
          %da
        %-  head
        %^    (dip:on-path (unit (each page @uvI)))
            fan.u.ski
          ~
        |=  [res=(unit (each page @uvI)) @ud =@da val=(each page @uvI)]
        ^-  [new=(unit [@da _val]) stop=? res=(unit _val)]
        :-  `[da val]
        ?:((lte da p.r.bem) |/`val &/res)
      ==
    ?.  ?=([~ %& *] res)  ~        :: change ~ to [~ ~] to fix
    ``p.u.res(q !>(q.p.u.res))
```
It's still unclear to me why the issue arises at all, but changing `~` to `[~ ~]` on the final branch resolves it.

**Post-fix:**

Non-tombstoned response: `dat=[p=path q=[~ [p=%atom q=123.456.789]]]`

Tombstoned response: `dat=[p=path q=~]`

Note here that `q`, which is a `(unit (cask))`, now returns `~` in the tombstone case. Perhaps that's okay. I conceive two alternatives:

a) `dat=[p=path q=[~ ~]`

Conforms with unitized structure of the non-tombstoned case. I'm unsure how to rewrite the handler to produce this, due compiler-enforced typing.

b) `dat=[p=path q=[~ tombstone-hash]`

Clay remembers hashes of tombstoned values. If we wanted to serve them, I think the response mold would have to be reconsidered, since the current `(unit (cask))` would require a reserved term like `%$` to express tombstones, i.e.: `[~ [%$ 0vn.ebj2e]]`

**Local crash**

Locally executed %gx reads crash on tombstones, and the proposed fix does not solve this:

`.^(noun %gx /(scot %p our)/app/0//i-am-valid)` - produces value

`.^(noun %gz /(scot %p our)/app/0//i-am-valid)` - produces hash

`.^(noun %gx /(scot %p our)/app/0//i-am-tombstoned)` - **crashes**

`.^(noun %gz /(scot %p our)/app/0//i-am-tombstoned)` - produces hash

It's possible to work around not having the Pine protocol, especially if .^ doesn't crash on tombstones. If .^ is unreliable for local fetching, userspace devs would need to implement a secondary system to track path-revisions in-app. I believe fixing .^ here would be quite valuable in the interim between now and Pine.
